### PR TITLE
[Enhancement] Add Scale support for SideMenuView

### LINKED ISSUE #1007: [Enhancement] Add Scale support for SideMenuView
## Summary

The `SideMenuView` is a nice View and I think we can make it a bit more Fancy by adding a scale animation to the main-view.
I'd like to get something like what @ahoefling shows in this BlogPost: https://www.andrewhoefling.com/Blog/Post/xamarin-forms-fancy-flyout-menu

Fancy Flyout

## API Changes

There is already a `Shift` property on the `SideMenuView` which is used to translate the menu with an animation, so we would only need a `Scale` property which is also taken into account when executing the animation.
This should be a small change and shouldn't have any implications.

## Intended Use Case
More fancy `SideMenuView`

## Who Will Do The Work?

- [ ] I am willing to take this on myself
- [x] Just putting this out there for someone to pick up

### PR DESCRIPTION
**Pull request by AndreiMisiukevich**
### Description of Change ###
Make it possible to scale the main view during the interaction.

### Bugs Fixed ###
- Fixes #1007

### API Changes ###
Added: 
 - `double SideMenuView.MainViewScaleFactor` (attached property)

### Behavioral Changes ###
None

### PR Checklist ###
- [ ] Has tests (if omitted, state reason in description)
- [X] Has samples (if omitted, state reason in description)
- [X] Rebased on top of main at time of PR
- [X] Changes adhere to coding standard
- [ ] Updated [documentation](https://github.com/MicrosoftDocs/xamarin-communitytoolkit)

### CODE REVIEW

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Views/SideMenuView/SideMenuView.shared.cs [hunk 1]
```diff
@@ -87,6 +87,9 @@ public class SideMenuView : BaseTemplatedView<AbsoluteLayout>
 		public static readonly BindableProperty MenuGestureEnabledProperty
 			= BindableProperty.CreateAttached(nameof(GetMenuGestureEnabled), typeof(bool), typeof(SideMenuView), true);
 
+		public static readonly BindableProperty MainViewScaleFactorProperty
+			= BindableProperty.CreateAttached(nameof(GetMainViewScaleFactor), typeof(double), typeof(SideMenuView), 1.0);
```
Comment (pictos) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Views/SideMenuView/SideMenuView.shared.cs:91]: Didn't know that Forms remove the Get prefix

**Review by pictos — APPROVED** [review]
Well done!

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Views/SideMenuView/SideMenuView.shared.cs (hunk 1 above)
Comment (AndreiMisiukevich) [reply src/CommunityToolkit/Xamarin.CommunityToolkit/Views/SideMenuView/SideMenuView.shared.cs:91]: What do you mean?)

Comment (pictos) [reply src/CommunityToolkit/Xamarin.CommunityToolkit/Views/SideMenuView/SideMenuView.shared.cs:91]: @AndreiMisiukevich There you typed `nameof(GetMainViewScaleFactor)` and on xaml you use `SideMenuView.MainViewScaleFactor` there's no `Get` prefix

Comment (AndreiMisiukevich) [reply src/CommunityToolkit/Xamarin.CommunityToolkit/Views/SideMenuView/SideMenuView.shared.cs:91]: @pictos I believe Xaml works with the name of BindableProperty. In this case, it's **MainViewScaleFactorProperty**
And it just removes postfix "Property".

Comment (pictos) [reply src/CommunityToolkit/Xamarin.CommunityToolkit/Views/SideMenuView/SideMenuView.shared.cs:91]: @AndreiMisiukevich makes sense...

**Pull request merged**